sdk/formrecognizer/Azure.AI.FormRecognizer/src/autorest.md

```diff
@@ -148,27 +156,37 @@ directive:
     where: $.definitions.DocumentSelectionMarkState
     transform: >
       $["x-ms-enum"].name = "V3SelectionMarkState";
+
 # Move enums to the DocumentAnalysis namespace
   - from: swagger-document
     where: $.definitions..properties.*
     transform: >
       if ($.enum) {
         $["x-namespace"] = "Azure.AI.FormRecognizer.DocumentAnalysis"
       }
+
 # Move QueryStringIndexType to the DocumentAnalysis namespace
   - from: swagger-document
     where: $.parameters.QueryStringIndexType
     transform: >
       $["x-namespace"] = "Azure.AI.FormRecognizer.DocumentAnalysis"
+
 # Setting these input IDs' formats to `uuid` moves the format validation to the client side, changing the type of the Exception thrown in case of error (breaking change).
   - from: swagger-document
     where: $.parameters
     transform: >
       $.PathOperationId.format = undefined;
       $.PathResultId.format = undefined;
+
 # Setting the Operation-Location format to `uuid` produces code that fails during runtime. The extracted header (a string) cannot be converted to `Uri` directly.
   - from: swagger-document
     where: $.paths..Operation-Location
     transform: >
       $.format = undefined;
+
+# Removing new properties in ResourceDetails from the list of required properties, otherwise deserialization breaks when calling older service versions.
+  - from: swagger-document
+    where: $.definitions.ResourceDetails
+    transform: >
+      $.required = [ "customDocumentModels" ];
```

Comment:
Would it be "safer" to specifically delete the properties that we don't want instead of overriding the entire array? I think something like this would work:

```yaml
  - from: swagger-document
    where: $.definitions.ResourceDetails.required
    transform: >
      $.splice($.indexOf("customNeuralDocumentModelBuilds"), 1);
      return $;
```

My reasoning is that this way we would know when new required properties are added (instead of them just getting swallowed by the transform), and then we can make an intentional decision about how we want to handle them. Depending on the answer to my previous comment (about if we need to add convenience constructors that take the new required properties as a parameters), having something like this might be important.

Reply:
Based on my response to your previous comment, our deserialization code will always pick the constructor that takes all parameters and makes no checks, so we wouldn't be aware that a new required property has been added.

I like the idea of minimizing changes to the swagger, though. I'll take your suggestion and update the autorest file.